modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/wstep/impl/CertificateEnrollmentServiceImpl.java

```diff
@@ -303,14 +311,35 @@ public String prepareWapProvisioningXML(
             Node pollValue = pollingAttributes.getNamedItem(PluginConstants.CertificateEnrolment.VALUE);
             pollValue.setTextContent(pollingFrequency);
             if (log.isDebugEnabled()) {
-                log.debug("Username: " + username + "Password: " + requestSecurityTokenResponse);
+                log.debug("Username: " + userName + "Password: " + requestSecurityTokenResponse);
             }
             wapProvisioningString = convertDocumentToString(document);
-
-            //Generic exception is caught here as there is no need of taking different actions for
-            //different exceptions.
-        } catch (Exception e) {
-            String msg = "Problem occurred with wap-provisioning.xml file.";
+        } catch (ParserConfigurationException e) {
+            String msg = "Problem occurred in parsing wap-provisioning.xml file.";
+            log.error(msg, e);
+            throw new WAPProvisioningException(msg, e);
+        } catch (DeviceManagementException e) {
+            String msg = "Error occurred in while getting CA and Root certificates.";
+            log.error(msg, e);
+            throw new WindowsDeviceEnrolmentException(msg, e);
+        } catch (NoSuchAlgorithmException | CertificateEncodingException e) {
+            String msg = "Error occurred in while encoding certificates.";
+            log.error(msg, e);
+            throw new WindowsDeviceEnrolmentException(msg, e);
+        } catch (UnsupportedEncodingException e) {
+            String msg = "Error occurred in while encoding wap-provisioning file.";
+            log.error(msg, e);
+            throw new WindowsDeviceEnrolmentException(msg, e);
+        } catch (SAXException e) {
+            String msg = "Error occurred in while parsing wap-provisioning.xml file.";
+            log.error(msg, e);
+            throw new WAPProvisioningException(msg, e);
+        } catch (TransformerException e) {
+            String msg = "Error occurred in while transforming wap-provisioning.xml file.";
+            log.error(msg, e);
```

Comment:
This method could be private. once this method becomes private, log messages become redundant